cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -338,47 +339,53 @@ public Future<CruiseControlResponse> getUserTaskStatus(Reconciliation reconcilia
-                                    CruiseControlUserTaskStatus taskStatus = CruiseControlUserTaskStatus.lookup(taskStatusStr);
-                                    switch (taskStatus) {
-                                        case ACTIVE:
-                                            // If the status is ACTIVE there will not be a "summary" so we skip pulling the summary key
-                                            break;
-                                        case IN_EXECUTION:
-                                            // Tasks in execution will be rebalance tasks, so their original response will contain the summary of the rebalance they are executing
-                                            // We handle these in the same way as COMPLETED tasks so we drop down to that case.
-                                        case COMPLETED:
-                                            // Completed tasks will have the original rebalance proposal summary in their original response
-                                            JsonObject originalResponse = (JsonObject) Json.decodeValue(jsonUserTask.getString(
-                                                    CruiseControlRebalanceKeys.ORIGINAL_RESPONSE.getKey()));
-                                            statusJson.put(CruiseControlRebalanceKeys.SUMMARY.getKey(),
-                                                    originalResponse.getJsonObject(CruiseControlRebalanceKeys.SUMMARY.getKey()));
-                                            // Extract the load before/after information for the brokers
-                                            statusJson.put(
-                                                    CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey(),
-                                                    originalResponse.getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey()));
-                                            statusJson.put(
-                                                    CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey(),
-                                                    originalResponse.getJsonObject(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey()));
-                                            break;
-                                        case COMPLETED_WITH_ERROR:
-                                            // Completed with error tasks will have "CompletedWithError" as their original response, which is not Json.
-                                            statusJson.put(CruiseControlRebalanceKeys.SUMMARY.getKey(), jsonUserTask.getString(CruiseControlRebalanceKeys.ORIGINAL_RESPONSE.getKey()));
-                                            break;
-                                        default:
-                                            throw new IllegalStateException("Unexpected user task status: " + taskStatus);
+                                    if (userTasks.isEmpty()) {
```

Comment:
Not sure we are tackling this right. I mean, the CC tasks could have multiple tasks but the one we are looking for, for any reasons I can't imagine right now, could have been removed by CC from the tasks list while others are still there.
If this happens, we are assuming the list not empty and then getting just the first task in the list which could even not be the one we are looking for.
Thinking more, it could have been wrong even before.
What I am saying is that we should look for the requested `userTaskId` in this list and if not found returning the empty status.
@kyguy @fvaleri wdyt?

Reply:
I think this is correct, because the  response is for a single `userTaskId`. If it is empty, then Cruise Control restarted, or the task response expired, or it was deleted because the internal cache was full.

This is indeed a minor issue with the Topic Operator, that has the ability to do batch requests to the `user_tasks` endpoint. I say minor, because it will simply results in a very little overhead caused by duplicates requests that would be no-ops (RF was already changed). Anyway, I'm taking note and will fix it soon.

Reply:
> I think this is correct, because the response is for a single userTaskId.

Genuine question. Are you sure about that? Should we look at CC code as well? I am wondering why if it's the response to a single task, it contains a list of `userTasks` instead of just the `Status` for that task or an empty JSON. I am worried that CC could always return the entire list.

Reply:
I'm quite sure. I tested that for the RF change feature. The response contains a task list because the endpoint supports batch requests, that we don't leverage in the Rebalance operator.

Reply:
See here https://github.com/linkedin/cruise-control/blob/main/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/sync/UserTasksRequest.java#L47 and https://github.com/linkedin/cruise-control/blob/main/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java#L510 

isn't it getting ALL user tasks?

Reply:
Ok wait wait, the endpoint by default returns all user tasks but luckily in our code we are using this ...

```java
if (userTaskId != null) {
            pathBuilder.withParameter(CruiseControlParameters.USER_TASK_IDS, userTaskId);
        }
```

we are setting the `user_task_ids` parameter on the query string. But without it, it would return all tasks as I said.

See also https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-user-request-result

Reply:
Yep, that's correct.

Reply:
Do we need some change w.r.t to this?  Following the discussion it sounds we agree that its correct this way

Reply:
Yes that's correct